src/l500/l500-device.h

```diff
@@ -108,6 +112,12 @@ namespace librealsense
 
         std::vector< calibration_change_callback_ptr > _calibration_change_callbacks;
         platform::usb_spec _usb_mode;
+
+        std::mutex _temperature_mutex;
+        std::atomic_bool _temperature_fetcher_on { false };  // Indicates if the temperature thread is working
+        std::atomic_bool _temperature_from_fetcher{ false }; // Indicates it the temperature value should be taking from the fetcher or by FW command
```

Comment:
Needed only if you don't use shared_ptr, no?

Reply:
I like `_have_temperatures` -- if `true`, then the `_temperatures` are valid and up-to-date, no?

Reply:
I renamed it.